src/components/Layout/RightPaneLayout/RightPaneContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactNode, useCallback, useContext, useMemo, useState } from "react";
+
+export interface OpenRightPaneOpts {
+  content: ReactNode;
+}
+
+export type RightPaneLayoutContextProps = {
+  openInPane: (opts: OpenRightPaneOpts) => void;
+  closePane: () => void;
+  clearPane: () => void;
+  isRightPaneOpen: Boolean;
+  rightPaneContent: ReactNode;
+};
+
+export const RightPaneContext = React.createContext<RightPaneLayoutContextProps>({
+  openInPane: () => {},
+  closePane: () => {},
+  clearPane: () => {},
+  isRightPaneOpen: false,
+  rightPaneContent: null,
+});
+
+export function RightPaneProvider({ children }: { children: ReactNode }) {
+  const [rightPaneContent, setRightPaneContent] = useState<ReactNode>(undefined);
```

Comment:
Suggestion: Maybe add an in-source comment that we track the pane content separate from is open so that we can animate away the pane content even though the pane is now closed?

Just thinking I'd probably re-wonder this reading/maintaining the code a few months down the road.

Reply:
Added note to this for future reference.